### PR TITLE
Add test for RTCPeerConnection.generateCertificate

### PR DESCRIPTION
**Pull request by soareschen**
This adds basic testing to the `RTCPeerConnection.generateCertificate()` method. It mainly covers:

- Test for compulsory parameters to be supported
- Test for `expiry` argument provided.

The `getFingerprints` and `getAlgorithm` methods will be tested separately in PR for `RTCCertificate`.

### CODE REVIEW

File: webrtc/RTCPeerConnection-generateCertificate.html [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+   */
+  promise_test(t =>
+    RTCPeerConnection.generateCertificate({
+      name: 'RSASSA-PKCS1-v1_5',
+      modulusLength: 2048,
+      publicExponent: new Uint8Array([1, 0, 1]),
+      hash: 'SHA-256'
+    }).then(cert => {
+      assert_true(cert instanceof RTCCertificate,
+        'Expect cert to be instance of RTCCertificate');
```
Comment (henbos) [inline webrtc/RTCPeerConnection-generateCertificate.html]: Here and below: you could assert that the generated cert is not an expired one.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/238267207)

# Firefox (nightly) #
Testing web-platform-tests at revision 3e20e226062d290815ec478a380c42a504701128
Using browser at version BuildID 20170531100318; SourceStamp 7b8937970f9ca85db88cb2496f2112175fd847c8
Starting 10 test iterations
## Unstable results ##


|                         Test                         |                                    Subtest                                    |          Results           |                                                                                                                                                                                                                                                                                                                                      Messages                                                                                                                                                                                                                                                                                                                                     |
|------------------------------------------------------|-------------------------------------------------------------------------------|----------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `/webrtc/RTCPeerConnection-generateCertificate.html` | `generateCertificate() with 0 expires parameter should generate expired cert` | **FAIL: 8/10, PASS: 2/10** | `assert_less_than: expected a number less than 1496309914338 but got 1496309914338;assert_less_than: expected a number less than 1496309928738 but got 1496309928738;assert_less_than: expected a number less than 1496309910989 but got 1496309910989;assert_less_than: expected a number less than 1496309938937 but got 1496309938937;assert_less_than: expected a number less than 1496309932166 but got 1496309932166;assert_less_than: expected a number less than 1496309921155 but got 1496309921155;assert_less_than: expected a number less than 1496309925355 but got 1496309925355;assert_less_than: expected a number less than 1496309917714 but got 1496309917714` |

## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6124/webrtc/RTCPeerConnection-generateCertificate.html">/webrtc/RTCPeerConnection-generateCertificate.html</a></summary>


|                                          Subtest                                           |          Results           |                                                                                                                                                                                                                                                                                                                                      Messages                                                                                                                                                                                                                                                                                                                                     |
|--------------------------------------------------------------------------------------------|----------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
|                                                                                            | OK                         |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
| `generateCertificate() with compulsary RSASSA-PKCS1-v1_5 parameters should succeed`        | PASS                       |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
| `generateCertificate() with compulsary ECDSA parameters should succeed`                    | PASS                       |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
| `generateCertificate() with invalid string algorithm should reject with NotSupportedError` | FAIL                       | `assert_throws: function "function() { throw e }" threw object "SyntaxError: An invalid or illegal string was specified" that is not a DOMException NotSupportedError: property "code" is equal to 12, expected 9`                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
| `generateCertificate() with invalid algorithm dict should reject with NotSupportedError`   | FAIL                       | `assert_throws: function "function() { throw e }" threw object "SyntaxError: An invalid or illegal string was specified" that is not a DOMException NotSupportedError: property "code" is equal to 12, expected 9`                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
| `generateCertificate() with valid expires parameter should succeed`                        | PASS                       |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
| `generateCertificate() with 0 expires parameter should generate expired cert`              | **FAIL: 8/10, PASS: 2/10** | `assert_less_than: expected a number less than 1496309914338 but got 1496309914338;assert_less_than: expected a number less than 1496309928738 but got 1496309928738;assert_less_than: expected a number less than 1496309910989 but got 1496309910989;assert_less_than: expected a number less than 1496309938937 but got 1496309938937;assert_less_than: expected a number less than 1496309932166 but got 1496309932166;assert_less_than: expected a number less than 1496309921155 but got 1496309921155;assert_less_than: expected a number less than 1496309925355 but got 1496309925355;assert_less_than: expected a number less than 1496309917714 but got 1496309917714` |
| `generateCertificate() with invalid range for expires should reject with TypeError`        | PASS                       |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
| `generateCertificate() with invalid type for expires should reject with TypeError`         | PASS                       |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/238267208)

# Sauce (safari) #
Testing web-platform-tests at revision 3e20e226062d290815ec478a380c42a504701128
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6124/webrtc/RTCPeerConnection-generateCertificate.html">/webrtc/RTCPeerConnection-generateCertificate.html</a></summary>


|                                          Subtest                                           | Results |                 Messages                 |
|--------------------------------------------------------------------------------------------|---------|------------------------------------------|
|                                                                                            | OK      |                                          |
| `generateCertificate() with compulsary RSASSA-PKCS1-v1_5 parameters should succeed`        | FAIL    | `Can't find variable: RTCPeerConnection` |
| `generateCertificate() with compulsary ECDSA parameters should succeed`                    | FAIL    | `Can't find variable: RTCPeerConnection` |
| `generateCertificate() with invalid string algorithm should reject with NotSupportedError` | FAIL    | `Can't find variable: RTCPeerConnection` |
| `generateCertificate() with invalid algorithm dict should reject with NotSupportedError`   | FAIL    | `Can't find variable: RTCPeerConnection` |
| `generateCertificate() with valid expires parameter should succeed`                        | FAIL    | `Can't find variable: RTCPeerConnection` |
| `generateCertificate() with 0 expires parameter should generate expired cert`              | FAIL    | `Can't find variable: RTCPeerConnection` |
| `generateCertificate() with invalid range for expires should reject with TypeError`        | FAIL    | `Can't find variable: RTCPeerConnection` |
| `generateCertificate() with invalid type for expires should reject with TypeError`         | FAIL    | `Can't find variable: RTCPeerConnection` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/238267216)

# Chrome (unstable) #
Testing web-platform-tests at revision 3e20e226062d290815ec478a380c42a504701128
Using browser at version 60.0.3112.7 dev
Starting 10 test iterations

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/238267217)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 3e20e226062d290815ec478a380c42a504701128

File: webrtc/RTCPeerConnection-generateCertificate.html [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    })
+  }, 'generateCertificate() with valid expires parameter should succeed');
+
+  promise_test(t => {
+    return RTCPeerConnection.generateCertificate({
+      name: 'ECDSA',
+      namedCurve: 'P-256',
+      expires: 0
+    }).then(cert => {
+      assert_less_than(cert.expires, Date.now());
```
Comment (henbos) [inline webrtc/RTCPeerConnection-generateCertificate.html]: Prefer assert_less_than_equal

**Review by henbos — APPROVED** [review]
lgtm with comments addressed.

**Review by rwaldron — APPROVED** [review]
Updated webrtc spec annotations:  https://rwaldron.github.io/webrtc-pc/#sec.cert-mgmt

**Pull request merged**